daemons/fenced/fenced_commands.c

```diff
@@ -1197,9 +1177,63 @@ stonith_device_register(xmlNode * msg, const char **desc, gboolean from_cib)
     stonith_device_t *dup = NULL;
     stonith_device_t *device = build_device_from_xml(msg);
     guint ndevices = 0;
+    int rv = pcmk_ok;
 
     CRM_CHECK(device != NULL, return -ENOMEM);
 
+    /* do we have a watchdog-device? */
+    if (pcmk__str_eq(device->id, STONITH_WATCHDOG_ID, pcmk__str_none) ||
+        pcmk__str_any_of(device->agent, STONITH_WATCHDOG_AGENT,
+                     STONITH_WATCHDOG_AGENT_INTERNAL, NULL)) do {
+        if (stonith_watchdog_timeout_ms <= 0) {
+            crm_err("Ignoring watchdog fence device without "
+                    "stonith-watchdog-timeout set.");
+            rv = -ENODEV;
+            /* fall through to cleanup & return */
+        } else if (!pcmk__str_any_of(device->agent, STONITH_WATCHDOG_AGENT,
+                                 STONITH_WATCHDOG_AGENT_INTERNAL, NULL)) {
+            crm_err("Ignoring watchdog fence device with "
+                    "agent '%s'!='" STONITH_WATCHDOG_AGENT
+                    "' || '" STONITH_WATCHDOG_AGENT_INTERNAL "'.",
+                    device->agent?device->agent:"");
```

Comment:
Given where the quotation marks are this shouldn't expose programming symbols but rather the agent-names themselves - right? So I guess "fence_watchdog" should be fine because it actually is what the user enters in pcs. But yes "#watchdog" is possibly not the best idea. But I guess we could drop that anyway as it shouldn't happen on a device visible to the user.
Or stay more cryptic but universally true: "Ignoring watchdog fence device with unknown agent unequal " STONITH_WATCHDOG_AGENT "."

Reply:
Right, I meant the "!=". Something like

    "Ignoring fence device 'watchdog' because agent '%s' is not " STONITH_WATCHDOG_AGENT

Reply:
Quite close to what I had chose I guess ;-)